### ISSUE TITLE
[$250] [OldDot & NewDot] Allow disabling auto renew freely if you haven't been billed at least once. 

### ISSUE THREAD

**Issue by trjExpensify** [Expensify/App#53382]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.69-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @trjExpensify 
**Slack conversation**: [#convert thread](https://expensify.slack.com/archives/C07HPDRELLD/p1732232110750859)

## Action Performed:

**NewDot**

1. Go to new.expensify.com
2. Sign-up for a new account 
3. Choose "Manage my team's expenses" and complete the onboarding modal
4. Go to Settings > Subscription > Subscription details 
5. Disable the auto renew toggle

**OldDot** 
1. Go to expensify.com
2. Choose "Control expenses for a larger organization"
3. Go to Settings > Workspaces > Group > Subscription 
4. Disable the auto renew toggle

## Expected Result:
_This is an improvement._

If the user hasn't been billed at least once, toggling off auto-renew should be allowed without having to complete the retention survey. 

## Actual Result:
The retention survey is shown to the user, which is confusing because none of the options make sense to a new user who hasn't been billed yet. 

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
FS [here](https://fsty.io/v/N9DvpQsT) (internal)

<img width="1047" alt="image" src="https://github.com/user-attachments/assets/d64e6bab-a5a7-41f3-8ca0-ee193b7ea1f2">

![image (23)](https://github.com/user-attachments/assets/6129a81c-4501-4116-b9ee-a0addb258bf0)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866071159246990575</li>
        <li>Upwork Job ID: 1866071159246990575</li>
        <li>Last Price Increase: 2024-12-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by saifelance** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue.

Users who have not been billed yet encounter a retention survey when attempting to disable the auto-renew feature. This is unnecessary and confusing because the retention survey options do not make sense for new users who have not incurred any charges.

### What is the root cause of that problem?

The current logic for disabling auto-renew does not differentiate between users who have been billed and those who have not. As a result, all users are shown the retention survey regardless of their billing history.

### What changes do you think we should make in order to solve the problem?

`src/pages/settings/Subscription/SubscriptionSettings/index.tsx`

### Determine Billing Status:
Add a check to determine if the user has been billed at least once.
The `privateSubscription` object should include a property (`e.g., hasBeenBilled`) to track whether the user has been billed. This information can be derived from the server or the subscription details.

### Modify handleAutoRenewToggle:
Update the logic to bypass the retention survey if the user hasn't been billed yet:

```
const handleAutoRenewToggle = () => {
    if (!privateSubscription?.autoRenew) {
        Subscription.updateSubscriptionAutoRenew(true);
        return;
    }
    
    if (!privateSubscription?.hasBeenBilled) {
        // Allow disabling auto-renew freely if user hasn't been billed
        Subscription.updateSubscriptionAutoRenew(false);
        return;
    }

    // Navigate to the retention survey if the user has been billed
    Navigation.navigate(ROUTES.SETTINGS_SUBSCRIPTION_DISABLE_AUTO_RENEW_SURVEY);
};

```

**Comment by trjExpensify** [comment]
@mananjadhav @rushatgabhane @mountiny I labelled this as internal, because the retention survey is also shown in this case on OldDot, and I suspect that simply skipping the survey isn't enough to actually disable autoRenew properly. Keen for a second opinion on that.

**Comment by mountiny** [comment]
Makes sense to me but I think the the contributors can also check out their solution if it works with purely frontend change. I am not sure if we require the the survey response in BE

**Comment by trjExpensify** [comment]
@Shahidullah-Muffakir maybe you can check then. :)

**Comment by Shahidullah-Muffakir** [comment]
@trjExpensify @mountiny I believe FE changes should be enough, as we can use `account?.hasPurchases` for billing history the same way, we used it here:https://github.com/Expensify/App/issues/51892

https://github.com/user-attachments/assets/4c09e3bf-5a82-4246-9187-7de471cc0d35

**Comment by trjExpensify** [comment]
Can you confirm that properly disables auto-renew? I guess, log-out and back in, or refresh or something?

**Comment by Shahidullah-Muffakir** [comment]
> Can you confirm that properly disables auto-renew? I guess, log-out and back in, or refresh or something?

Yes, I tested it, and it's working properly!

https://github.com/user-attachments/assets/ac02b7a9-0f1e-489c-ade7-70ef3147fd5d

**Comment by trjExpensify** [comment]
Great stuff! Let's get that moving forward then. @akinwale can you review the proposals, please?

**Comment by akinwale** [comment]
After reviewing both proposals, we can move forward with @Shahidullah-Muffakir's [proposal](https://github.com/Expensify/App/issues/53382#issuecomment-2511750718) since it was posted first and it reuses an approach that already exists in the codebase.

🎀👀🎀 C+ reviewed.

**Comment by akinwale** [comment]
This was deployed to production on 2024-12-18, so it will be due for payment on 2024-12-25.

Happy holidays!

**Comment by akinwale** [comment]
This was deployed to production on [2024-12-18](https://github.com/Expensify/App/issues/54111#issuecomment-2551261636), so it will be due for payment on 2024-12-25.

Happy holidays!

**Comment by Shahidullah-Muffakir** [comment]
Friendly bump for pending payment.

**Comment by bfitzexpensify** [comment]
Payment complete to @Shahidullah-Muffakir 

@akinwale due $250 via ND manual request for C+ work. 

Will leave this open for you to close @trjExpensify

**Comment by trjExpensify** [comment]
Thanks for handling that, @bfitzexpensify. @techievivek will you take on the OldDot side of this issue?

**Comment by JmillsExpensify** [comment]
$250 approved for @akinwale

**Comment by trjExpensify** [comment]
Bump [on this](https://github.com/Expensify/App/issues/53382#issuecomment-2569094233), @techievivek.

**Comment by techievivek** [comment]
Aaah, sorry, I missed the ping, I can take a look into it tomorrow.

**Comment by trjExpensify** [comment]
Thank you!

**Comment by amyevans** [comment]
I saw this near the top of the Hot Picks list - @techievivek you got it covered or are you looking to hand off?

**Comment by techievivek** [comment]
Hi, @amyevans, please feel free to take over, I have few other pressing tasks on my plate. Thanks.

**Comment by amyevans** [comment]
PRs are into review.

Auth PR: https://github.com/Expensify/Auth/pull/13874
Web PR: https://github.com/Expensify/Web-Expensify/pull/45183

**Comment by amyevans** [comment]
The OldDot portion is live now too so I believe we're all set to close!
